src/js.js

```diff
@@ -1,115 +1,134 @@
     case 'star':
-      counter.innerHTML = addCommas(obj.data.stargazers_count);
-      counter.setAttribute('aria-label', counter.innerHTML + ' stargazers' + labelSuffix);
+      mainButton.className += ' github-stargazers';
+      text.textContent = 'Star';
+      counter.href = REPO_URL + '/stargazers';
       break;
     case 'fork':
-      counter.innerHTML = addCommas(obj.data.network_count);
-      counter.setAttribute('aria-label', counter.innerHTML + ' forks' + labelSuffix);
+      mainButton.className += ' github-forks';
+      text.textContent = 'Fork';
+      button.href = REPO_URL + '/fork';
+      counter.href = REPO_URL + '/network';
       break;
     case 'follow':
-      counter.innerHTML = addCommas(obj.data.followers);
-      counter.setAttribute('aria-label', counter.innerHTML + ' followers' + labelSuffix);
+      mainButton.className += ' github-me';
+      text.textContent = 'Follow @' + user;
+      button.href = GH_URL + user;
+      counter.href = GH_URL + user + '/followers';
+      break;
+    case 'sponsor':
+      mainButton.className += ' github-me';
+      text.textContent = 'Sponsor @' + user;
+      button.href = GH_URL + user + '/sponsorship';
+      counter.href = GH_URL + user + '/followers';
       break;
   }
 
-  // Show the count if asked
-  if (count === 'true' && counter.innerHTML !== 'undefined') {
-    counter.style.display = 'block';
-  }
-}
-
-// Set href to be URL for repo
-button.href = 'https://github.com/' + user + '/' + repo + '/';
+  button.setAttribute('aria-label', text.textContent + labelSuffix);
 
-// Add the class, change the text label, set count link href
-switch (type) {
-  case 'watch':
-    if (v === '2') {
-      mainButton.className += ' github-watchers';
-      text.innerHTML = 'Watch';
-      counter.href = 'https://github.com/' + user + '/' + repo + '/watchers';
-    } else {
-      mainButton.className += ' github-stargazers';
-      text.innerHTML = 'Star';
-      counter.href = 'https://github.com/' + user + '/' + repo + '/stargazers';
-    }
-    break;
-  case 'star':
-    mainButton.className += ' github-stargazers';
-    text.innerHTML = 'Star';
-    counter.href = 'https://github.com/' + user + '/' + repo + '/stargazers';
-    break;
-  case 'fork':
-    mainButton.className += ' github-forks';
-    text.innerHTML = 'Fork';
-    button.href = 'https://github.com/' + user + '/' + repo + '/fork';
-    counter.href = 'https://github.com/' + user + '/' + repo + '/network';
-    break;
-  case 'follow':
-    mainButton.className += ' github-me';
-    text.innerHTML = 'Follow @' + user;
-    button.href = 'https://github.com/' + user;
-    counter.href = 'https://github.com/' + user + '/followers';
-    break;
-}
-button.setAttribute('aria-label', text.innerHTML + labelSuffix);
-
-// Change the size
-if (size === 'large') {
-  mainButton.className += ' github-btn-large';
-}
+  // Change the size
+  if (size === 'large') {
+    mainButton.className += ' github-btn-large';
```

Comment:
I prefer when `classList` API is used, is it a matter of compatibility?

Reply:
This is how it was and I haven't change it yet because initially I didn't plan to drop old IE support. But it's in my TODO list. Note that this is the fastest way to add classes :P